### PR TITLE
Fix some flaky tests

### LINKED ISSUE #1066: test_can_generate_interval_endpoints is flaky
```
=================================== FAILURES ===================================
_ test_can_generate_interval_endpoints[-2.22507385851e-308-2.22507385851e-308] _
[gw1] linux2 -- Python 2.7.14 /usr/bin/python2
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.16-py2/tests/cover/test_float_nastiness.py", line 84, in test_can_generate_interval_endpoints
    find(interval, lambda x: x == r, settings=settings(max_examples=10000))
  File "/build/python-hypothesis/src/hypothesis-python-3.44.16-py2/src/hypothesis/core.py", line 1128, in find
    raise NoSuchExample(get_pretty_function_description(condition))
NoSuchExample: No examples found of condition lambda x: x == r
=============================== warnings summary ===============================
tests/numpy/test_gen_data.py::test_can_minimize_float_arrays
  /usr/lib/python2.7/site-packages/numpy/core/_methods.py:32: RuntimeWarning: invalid value encountered in reduce
    return umr_sum(a, axis, dtype, out, keepdims)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
======== 1 failed, 2380 passed, 6 skipped, 1 warnings in 100.77 seconds ========
```

Environment:
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-3.3.2, py-1.5.0, pluggy-0.6.0
rootdir: /build/python-hypothesis/src/hypothesis-python-3.44.16-py2, inifile: tox.ini
plugins: hypothesis-3.44.16, xdist-1.22.0, forked-0.2, flaky-3.4.0
```

### PR DESCRIPTION
**Pull request by DRMacIver**
A quality test I'd noticed was flaky plus fixes #1066.

### CODE REVIEW

File: tests/quality/test_shrink_quality.py [hunk 1]
```diff
@@ -88,11 +88,13 @@ def test_minimize_3_set_of_tuples():
         lambda x: len(x) >= 2) == set(((0,), (1,)))
 
 
-def test_minimize_sets_of_sets():
+@pytest.mark.parametrize('seed', [11, 43, 49, 56, 65])
```
Comment (alexwlchan) [inline tests/quality/test_shrink_quality.py]: Are these seeds significant?

Comment (DRMacIver) [reply tests/quality/test_shrink_quality.py]: Not super significant. They're ones that previously failed when I ran with seeds of 0 to 100 (I've also run the tests in the new form from 0 to 100 and they pass all of them).

Comment (Zac-HD) [reply tests/quality/test_shrink_quality.py]: If the test now passes for any seed I'd rather not seed at all - just bump down the size.

File: tests/quality/test_shrink_quality.py [hunk 2]
```diff
@@ -90,9 +90,9 @@ def test_minimize_3_set_of_tuples():
 
 def test_minimize_sets_of_sets():
     elements = integers(1, 100)
-    size = 10
+    size = 8
     set_of_sets = minimal(
-        sets(frozensets(elements)), lambda s: len(s) >= size
+        sets(frozensets(elements), min_size=size),
```
Comment (alexwlchan) [inline tests/quality/test_shrink_quality.py]: nit: I think you could collapse this all onto a single line.

**Pull request merged**